src/ServerConfig.cc

```diff
@@ -587,3 +692,229 @@ const std::vector<std::string> &ServerConfig::LogRecordTopics() const
+  doc.Parse(_str.c_str());
+  return parsePluginsFromDoc(doc);
+}
+
+
+/////////////////////////////////////////////////
+std::list<ServerConfig::PluginInfo>
+ignition::gazebo::loadPluginInfo(bool _isPlayback)
+{
+  std::list<ServerConfig::PluginInfo> ret;
+
+  // 1. Check contents of environment variable
+  std::string envConfig;
+  bool configSet = ignition::common::env(gazebo::kServerConfigPathEnv,
+                                         envConfig);
+
+  if (configSet)
+  {
+    if (ignition::common::exists(envConfig))
+    {
+      // Parse configuration stored in environment variable
+      ret = ignition::gazebo::parsePluginsFromFile(envConfig);
+      if (ret.empty())
+      {
+        // This may be desired behavior, but warn just in case.
+        // Some users may want to defer all loading until later
+        // during runtime.
+        ignwarn << gazebo::kServerConfigPathEnv
+                << " set but no plugins found\n";
+      }
+      igndbg << "Loaded (" << ret.size() << ") plugins from file " <<
+        "[" << envConfig << "]\n";
+
+      return ret;
+    }
+    else
+    {
+      // This may be desired behavior, but warn just in case.
+      // Some users may want to defer all loading until late
+      // during runtime.
+      ignwarn << gazebo::kServerConfigPathEnv
+              << " set but no file found,"
+              << " no plugins loaded\n";
+      return ret;
+    }
+  }
+
+  std::string configFilename;
+  if (_isPlayback)
+  {
+    configFilename = "playback_server.config";
+  }
+  else
+  {
+    configFilename = "server.config";
+  }
+
+  std::string defaultConfig;
+  ignition::common::env(IGN_HOMEDIR, defaultConfig);
+  defaultConfig = ignition::common::joinPaths(defaultConfig, ".ignition",
+    "gazebo", configFilename);
+
+  if (!ignition::common::exists(defaultConfig))
+  {
+    auto installedConfig = ignition::common::joinPaths(
+        IGNITION_GAZEBO_SERVER_CONFIG_PATH,
+        configFilename);
+
+    if (!ignition::common::exists(installedConfig))
+    {
+      ignerr << "Failed to copy installed config [" << installedConfig
+             << "] to default config [" << defaultConfig << "]."
+             << "(file " << installedConfig << " doesn't exist)"
+             << std::endl;
+      return ret;
+    }
+    else if (!ignition::common::copyFile(installedConfig, defaultConfig))
+    {
+      ignerr << "Failed to copy installed config [" << installedConfig
+             << "] to default config [" << defaultConfig << "]."
+             << std::endl;
+      return ret;
+    }
+    else
+    {
+      ignmsg << "Copied installed config [" << installedConfig
+             << "] to default config [" << defaultConfig << "]."
+             << std::endl;
+    }
+  }
+
+  ret = ignition::gazebo::parsePluginsFromFile(defaultConfig);
+
+  if (ret.empty())
+  {
+    // This may be desired behavior, but warn just in case.
+    ignwarn << "Loaded IGN_HOME config, but no plugins found\n";
```

Comment:
I don't see the term `IGN_HOME` being used anywhere else in the code. How about printing `defaultConfig` here?